src/components/TagDropdown.tsx

```diff
@@ -0,0 +1,156 @@
+//SELECTION:
+//Once you click on a tag, it should be added to the product's description or
+//smth
+
+//Needs to be added:
+//1. When user is deleting word, filtered list goes back to normal
+//2. Input text field goes over "Add tag" if the tag is really really long
+
+interface TagDropdownProps {
+    tags: string[];
+}
+
+export default function TagDropdown(props: TagDropdownProps) {
+    const [display, setDisplay] = useState(false);
+    const [selectedTags, setSelectedTags] = useState<string[]>([]);
+    const [searchInput, setSearchInput] = useState("");
+    const [filteredTags, setFilteredTags] = useState(props.tags);
+
+    const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const input_text: string = event.target.value;
+        event.preventDefault();
+        setSearchInput(input_text);
+        if (input_text.length > 0) {
+            setFilteredTags(
+                props.tags.filter((tag) => {
+                    return tag.match(input_text);
+                })
+            );
+        } else {
+            setFilteredTags(props.tags);
+        }
+    };
+
+    // Handles actions when 'Enter' key is pressed
+    const handleEnter = (key_event: React.KeyboardEvent) => {
+        if (key_event.key == "Enter") {
+            addTag(searchInput);
+        }
+    }
+
+    const handleAdd = () => {
+        addTag(searchInput);
+    }
+
+    // Controls display of dropdown
+    const handleClick = () => {
+        if (!display) {
+            setDisplay(true);
+        }
+    };
+
+    // Controls tag selection
+    function handleSelectedTags(tag: string) {
+        // if tag is already selected, deselect it and remove from selectedTags
+        if (selectedTags.includes(tag)) {
+            const toRemove = selectedTags.indexOf(tag);
+            setSelectedTags(() => [
+                ...selectedTags.slice(0, toRemove),
+                ...selectedTags.slice(toRemove + 1),
+            ]);
+        } else {
+            setSelectedTags([...selectedTags, tag]);
+        }
+    }
+
+    function deselectTag(tag: string) {
+        const toRemove = selectedTags.indexOf(tag);
+        setSelectedTags(() => [
+            ...selectedTags.slice(0, toRemove),
+            ...selectedTags.slice(toRemove + 1),
+        ]);
+    }
+
+    function addTag(tag: string) {
+        // If tag is in props.tags, add tag to selectedTags array
+        // Else (new tag), add tag to props.tags (large tag array), probably requires request to db, then add tag to selectedTags
+        // ?? QUESTION: do we actually want a New tag to be added every time someone hits enter
+        if (!props.tags.includes(tag)) {
+            props.tags.push(tag);
+            // console.log("pushed tag");
+            // console.log(props.tags);
+        } 
+        setSelectedTags([...selectedTags, tag]);
+        // clear search bar
+        setSearchInput("");
+    }
+
+    return (
+        <div className="flex flex-col p-4 gap-[10px]">
```

Comment:
Currently, once a user opens the dropdown, there is no way to close it again. It's not an ideal solution, but I would suggest adding an `onMouseLeave` handler to this div and have it such that when the user moves their cursor away from the dropdown it collapses again.

Reply:
I made it so that when you click inside the search bar again, it closes the dropdown div. That's actually the behavior we had originally and I think it makes sense (and also only involved adding an `else` condition to the `handleClick` handler), but of course we can add an `onMouseLeave` handler instead, just request another change lol

Reply:
on second thought.. I think I'll implement the `onMouseLeave` instead